packages/mongo-nav/src/org-nav/OrgNav.tsx

```diff
@@ -255,62 +284,69 @@ export default function OrgNav({
                       {currentProjectName ?? 'None'}
                     </p>
                   </Menu>
-                </li>
+                )}
+              </IconButton>
 
-                <li role="none" className={supportContainer}>
-                  <OrgNavLink
-                    href={orgNav.support}
-                    isActive={activeNav === 'support'}
-                    data-testid="org-support"
-                  >
-                    Support
-                  </OrgNavLink>
-                </li>
-
-                <li role="none">
-                  <OrgNavLink
-                    href={orgNav.billing}
-                    isActive={activeNav === 'billing'}
-                    data-testid="org-billing"
-                  >
-                    Billing
-                  </OrgNavLink>
-                </li>
-              </>
-            )}
-          </ul>
+              <OrgNavLink
+                href={current && orgNav.support}
+                isActive={activeNav === 'support'}
+                loading={!current}
+                className={supportContainer}
+                data-testid="org-nav-support"
+              >
+                Support
+              </OrgNavLink>
+
+              {!onPremEnabled && (
+                <OrgNavLink
+                  href={current && orgNav.billing}
+                  isActive={activeNav === 'billing'}
+                  loading={!current}
+                  data-testid="org-nav-billing"
+                >
+                  Billing
+                </OrgNavLink>
+              )}
+            </>
+          )}
+        </>
+      )}
+
+      <div
+        className={css`
+          margin-left: auto;
+        `}
+      >
+        {onPremEnabled && onPremVersion && (
+          <span className={versionStyle} data-testid="org-nav-on-prem-version">
+            {onPremVersion}
+          </span>
         )}
-      </div>
-      <div>
+
         {!isMobile && (
           <OrgNavLink
             href={orgNav.allClusters}
             isActive={activeNav === 'allClusters'}
             className={rightLinkMargin}
-            data-testid="all-clusters-link"
+            data-testid="org-nav-all-clusters-link"
           >
             All Clusters
           </OrgNavLink>
         )}
 
-        {!isTablet && admin && (
+        {!isTablet && admin && !onPremEnabled && (
           <OrgNavLink
             href={orgNav.admin}
             isActive={activeNav === 'admin'}
             className={rightLinkMargin}
-            data-testid="admin-link"
+            data-testid="org-nav-admin-link"
           >
             Admin
           </OrgNavLink>
         )}
-
-        <UserMenu
-          account={account}
-          activeProduct={activeProduct}
-          urls={urls}
-          hosts={hosts}
-        />
       </div>
+
+      {renderedUserMenu}
```

Comment:
[test coverage] I'd suggest testing the following cases, particularly because we're very careful about avoiding UI bugs in Ops Manager (different release schedules, supporting multiple versions, etc., makes it tedious to get a bugfix out, and then the release may still not be installed by a group)

### When On Prem is enabled
- I see an on prem user menu
- I do not see the default user menu
- I see the on prem version
- I do not see a billing link

#### When admin is true as well
- I still do not see an admin link

### Default state
- I do not see the on prem user menu
- I do not see the on prem version

Reply:
- When onPrem is enabled: 
I already tested for lack of billing link and that the proper UserMenu is displayed
- Default
I already test for proper User Menu displayed

Will add tests for: 
- onPrem: version is displayed, admin not displayed even when true
- Default: do not see version

Reply:
Okay added!